libs/s25main/figures/nofGeologist.cpp

```diff
@@ -71,8 +72,18 @@ nofGeologist::nofGeologist(SerializedGameData& sgd, const unsigned obj_id)
 
     node_goal = sgd.PopMapPoint();
 
-    for(bool& found : resAlreadyFound)
-        found = sgd.PopBool();
+    if(sgd.GetGameDataVersion() < 6)
+    {
+        for(const auto res : helpers::enumRange<ResourceType>())
+        {
+            if(res != ResourceType::Fish && res != ResourceType::Water)
+                resAlreadyFound[res] = sgd.PopBool();
+        }
+    } else
+    {
+        for(bool& found : resAlreadyFound)
```

Comment:
can this work at all? isnt "resAlreadyFound" empty on load?

Reply:
ah forget it, its always filled. ok :-)

Reply:
It is a fixed size array ;)
We could however provide serialization function pairs for those to avoid such loops.